doc/changes.md

```diff
@@ -1,5 +1,85 @@
 # Changelog
 
+## Release v0.12.0 (2024/05/01)
+
+[![Digital Object Identifier for PyGMT v0.12.0](https://zenodo.org/badge/DOI/10.5281/zenodo.11062720.svg)](https://doi.org/10.5281/zenodo.11062720)
+
+### Highlights
+
+*
```

Comment:
Some potential highlights:
```suggestion
* 🎉 **Twelveth minor release of PyGMT** 🎉
* Almost all module wrappers (with a few exceptions) now have consistent behavior for table-like output ([#1318](https://github.com/GenericMappingTools/pygmt/issues/1318))
* Almost all module wrappers (with a few exceptions) now use GMT *virtual files* for output, without writing output to temporary files ([#2730](https://github.com/GenericMappingTools/pygmt/issues/2730)) 🚀
* Adopt [SPEC 0](https://scientific-python.org/specs/spec-0000/) policy for minimum supported versions of GMT, Python and other core dependencies 
```

Reply:
> * Almost all module wrappers (with a few exceptions) now use GMT *virtual files* for output, without writing output to temporary files ([#2730](https://github.com/GenericMappingTools/pygmt/issues/2730))

From a user's perspective, this might be confusing since they won't actually see the virtualfiles, but an object like `pandas.DataFrame` or `xarray.DataArray`. Maybe reword this to say that:

`* Almost all module wrappers (with a few exceptions) now use in-memory GMT *virtual files* instead of intermediate temporary files to improve performance ([#2730](https://github.com/GenericMappingTools/pygmt/issues/2730))`

Reply:
Sounds good. I've updated the suggested changed.